.github/workflows/sonarqube.yml

```
@@ -18,10 +18,25 @@
+          python-version: '3.12'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r backend/requirements.txt
+
+      - name: Run tests and generate coverage report
+        run: |
+          cd backend
+          pytest --cov=api --cov=backend --cov-report=xml:coverage.xml
       
```

YAMLlint (1.35.1)

[error] trailing spaces

(trailing-spaces)